// react-carousel-with-thumbnail/dist/style.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.ThumbnailWrapper = exports.NavigationWrapper = exports.ChildrenWrapper = exports.CarouselWrapper = void 0;

var _styledComponents = _interopRequireWildcard(require("styled-components"));

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = Object.defineProperty && Object.getOwnPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : {}; if (desc.get || desc.set) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj["default"] = obj; return newObj; } }

function _templateObject9() {
  var data = _taggedTemplateLiteral(["\n  width: ", ";\n  height: ", ";\n  position: relative;\n  * {\n    box-sizing: border-box;\n  }\n"]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteral(["\n  width: 100%;\n  height: ", ";\n  overflow: hidden;\n  display: flex;\n  justify-content: space-between;\n  margin-top: ", ";\n  .thumb {\n    width: ", ";\n    height: ", ";\n    overflow: hidden;\n    background: #fff;\n    filter: grayscale(100);\n    opacity: 0.6;\n    transition: 0.3s ease-in-out;\n    overflow: hidden;\n    cursor: pointer;\n    img,\n    div {\n      width: 100%;\n      height: 100%;\n      object-fit: cover;\n      transition: 0.3s ease-in-out;\n      position: absolute;\n      top: 0;\n      left: 0;\n    }\n    img {\n      position: absolute;\n      top: 0;\n      left: 0;\n    }\n    div > * {\n      display: none;\n    }\n    div > img {\n      display: block;\n    }\n    .carousel-mask {\n      width: 100%;\n      height: 100%;\n      position: absolute;\n      z-index: 0;\n      background: ", ";\n    }\n    &:hover {\n      div,\n      img {\n        transform: scale(1.1);\n      }\n    }\n    &:nth-child(", ") {\n      filter: grayscale(0);\n      opacity: 1;\n      div,\n      img {\n        transform: scale(1.1);\n      }\n    }\n  }\n"]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteral(["\n  width: 100%;\n  height: 50px;\n  display: flex;\n  align-items: center;\n  justify-content: space-between;\n  position: absolute;\n  padding: 0 20px;\n  top: 0;\n  right: 0;\n  left: 0;\n  bottom: 0;\n  margin: auto;\n  .cal-navigation-item {\n    border: 3px solid ", ";\n    width: 22px;\n    height: 22px;\n    transform: rotate(45deg);\n    cursor: pointer;\n    &.prev {\n      border-right: 0;\n      border-top: 0;\n    }\n    &.next {\n      border-left: 0;\n      border-bottom: 0;\n    }\n  }\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteral(["\n  left: ", ";\n  &.move {\n    animation: ", " 0.4s ease-in-out forwards;\n  }\n  &.back {\n    animation: ", " 0.4s ease-in-out forwards;\n  }\n  .carousel-mask {\n    width: 100%;\n    height: 100%;\n    position: absolute;\n    z-index: -1;\n    background: ", ";\n  }\n"]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["\n    0% {\n      left: calc(", " * (", "));\n    }\n    100% {\n      left: calc(", " * ", ");\n  "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n    0% {\n      left: calc(", " * (", " - 1));\n    }\n    100% {\n      left: calc(", " * (", " - 2));\n  "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n  display: flex;\n  justify-content: center;\n  align-items: center;\n  width: 100%;\n  height: ", ";\n  position: relative;\n  overflow: hidden;\n  .inner {\n    width: 100%;\n    height: 100%;\n    position: relative;\n    transition: 0.3s ease-in-out;\n    .slide-item {\n      width: 100%;\n      height: 100%;\n      display: flex;\n      justify-content: center;\n      align-items: center;\n      z-index: -1;\n      position: absolute;\n      background: #ccc;\n      top: 0;\n      overflow: hidden;\n      flex-direction: column;\n      color: #fff;\n      img {\n        width: 100%;\n        height: 100%;\n        object-fit: cover;\n        position: absolute;\n        top: 0;\n        left: 0;\n        z-index: -1;\n      }\n      &.move {\n        &:nth-child(1) {\n          animation: ", " 0.4s ease-in-out 0s forwards;\n        }\n      }\n      &.back {\n        &:nth-child(", ") {\n          animation: ", " 0.4s ease-in-out 0s forwards;\n        }\n      }\n    }\n  }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n    0% {\n      top: ", ";\n    }\n    50% {\n      top: ", ";\n      left: -100%;\n    }\n    100% {\n      top: 0px;\n      left: -100%;\n    }\n  "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n    0% {\n      top: ", ";\n    }\n    25% {\n      top: ", ";\n      left: calc(", " * (", "));\n    }\n    50% {\n      top: 0px;\n      left: calc(", " * (", "));\n    }\n    100% {\n      top: 0px;\n      left: calc(", " * (", "));\n    }\n  "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

/* ===============================================
#  Carousel Wrapper
=============================================== */
var forward = function forward(props) {
  return (0, _styledComponents.keyframes)(_templateObject(), props.height, props.height, props.width, props.length - 1, props.width, props.length - 1, props.width, props.length - 2);
};

var backward = function backward(props) {
  return (0, _styledComponents.keyframes)(_templateObject2(), props.height, props.height);
};

var CarouselWrapper = _styledComponents["default"].div(_templateObject3(), function (props) {
  return "calc(".concat(props.height, " - ").concat(props.thumbHeight, " - ").concat(props.margin, ")");
}, function (props) {
  return forward(props);
}, function (props) {
  return props.length;
}, function (props) {
  return backward(props);
});
/* ===============================================
#  Children Wrapper
=============================================== */


exports.CarouselWrapper = CarouselWrapper;

var move = function move(props) {
  return (0, _styledComponents.keyframes)(_templateObject4(), props.width, props.index, props.width, props.index);
};

var back = function back(props) {
  return (0, _styledComponents.keyframes)(_templateObject5(), props.width, props.index - 1, props.width, props.index);
};

var ChildrenWrapper = _styledComponents["default"].div(_templateObject6(), function (props) {
  return "calc(".concat(props.width, " * (").concat(props.index - 1, "))");
}, function (props) {
  return move(props);
}, function (props) {
  return back(props);
}, function (props) {
  return props.maskColor;
});
/* ===============================================
#  Navigation Wrapper
=============================================== */


exports.ChildrenWrapper = ChildrenWrapper;

var NavigationWrapper = _styledComponents["default"].div(_templateObject7(), function (props) {
  return props.navigationColor;
});
/* ===============================================
#  Thumbnail Wrapper
=============================================== */


exports.NavigationWrapper = NavigationWrapper;

var ThumbnailWrapper = _styledComponents["default"].div(_templateObject8(), function (props) {
  return props.thumbHeight;
}, function (props) {
  return props.margin;
}, function (props) {
  return "calc(100% / ".concat(props.length, " - ").concat(props.margin, ")");
}, function (props) {
  return props.thumbHeight;
}, function (props) {
  return props.maskColor;
}, function (props) {
  return props.current;
});
/* ===============================================
#  Whole Wrapper
=============================================== */


exports.ThumbnailWrapper = ThumbnailWrapper;

var Wrapper = _styledComponents["default"].div(_templateObject9(), function (props) {
  return props.width;
}, function (props) {
  return props.height;
});

var _default = Wrapper;
exports["default"] = _default;
